VueJS/index.js

```diff
@@ -1,13 +1,33 @@
 import List from './components/List.js'
 import ListItem from './components/ListItem.js'
 
+Vue.component('todo-list', List);
+
 new Vue({
   el: '#app',
   data: {
-    todos: [
+    idCounter: -1,
+    items: [
       { id: '1', message: 'Foo', },
       { id: '2', message: 'Bar', },
       { id: '3', message: 'Baz', }
     ]
+  },
+  methods:{
+    add: function(){
+      this.items.push({id: this.getId(), message:''});
+    },
+    save: function(item){
+        this.items[item.listIndex].message = item.message;
```

Comment:
Why do you set the property of an existing `todo` instead of replacing the entire todo with `item`?

Reply:
This is a violation of https://en.wikipedia.org/wiki/Law_of_Demeter because the component needs to know the structure of the `todo`, ie. the `message` property.